File: Algorithms/src/2559. Count Vowel Strings in Ranges/Code/solution.cpp
#include <vector>
#include <string>
#include <unordered_set>
using namespace std;

class Solution
{
public:
    vector<int> vowelStrings(vector<string> &words, vector<vector<int>> &queries)
    {
        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
        int n = words.size();
        vector<int> prefix(n, 0);

        // Precompute the prefix sum
        for (int i = 0; i < n; ++i)
        {
            if (vowels.count(words[i].front()) && vowels.count(words[i].back()))
            {
                prefix[i] = 1;
            }
            if (i > 0)
            {
                prefix[i] += prefix[i - 1];
            }
        }

        // Answer the queries
        vector<int> result;
        for (const auto &query : queries)
        {
            int l = query[0], r = query[1];
            result.push_back(prefix[r] - (l > 0 ? prefix[l - 1] : 0));
        }
        return result;
    }
};

File: Algorithms/src/2559. Count Vowel Strings in Ranges/Code/solution.js
var vowelStrings = function (words, queries) {
  const vowels = new Set(["a", "e", "i", "o", "u"]);
  const n = words.length;
  const prefix = Array(n).fill(0);

  // Precompute the prefix sum
  for (let i = 0; i < n; i++) {
    if (vowels.has(words[i][0]) && vowels.has(words[i][words[i].length - 1])) {
      prefix[i] = 1;
    }
    if (i > 0) {
      prefix[i] += prefix[i - 1];
    }
  }

  // Answer the queries
  return queries.map(([l, r]) => prefix[r] - (l > 0 ? prefix[l - 1] : 0));
};
